### DISCUSSION TITLE
Upsert in ActiveRecord

### DISCUSSION THREAD

**Discussion by scottix** [yiisoft/yii2#19367] [Ideas]
I see we have upsert in the QueryBuilder which works and we use it.

Would it be feasible to make it part of ActiveRecord

ideally
```php
$model = new Model();
$model->load($params);
$model->upsert();
```

**Comment by simialbi** [comment]
Doesn't `save` already do this?

**Reply by scottix** [reply]
save() in this context only attempts to insert new data. It doesn't have the facilities to handle typical what is called a conflict.

Upsert works at the database level to make sure it is atomic and what  to do if there is a conflict.

**Comment by bizley** [comment]
We definitely won't add to Yii 2 as this would be a totally new feature, but you are free to prepare a custom package handling it though, maybe someone indeed is looking for that functionality (I don't, and never did, and never heard of anyone looking for it). The problem here would be a validation because with typical use case for scenarios (create/update) you don't know which scenario should be used for it until you save it which is too late.
